services/analyzer/mongo-collections-analyzer.js

```diff
@@ -18,7 +18,8 @@ const mapReduceOptions = {
 // NOTICE: This code runs on the MongoDB side (mapReduce feature).
 //         The supported JS version is not the same than elsewhere.
 //         The code used here must work with MongoDB lower version supported.
-/* eslint-disable vars-on-top, no-var, no-undef, no-restricted-syntax */
+/* eslint-disable vars-on-top, no-var, no-undef, no-restricted-syntax,
+                  sonarjs/cognitive-complexity */
```

Comment:
I deactivated the `sonarjs/cognitive-complexity` rule as this function is really sensible.